### PR TITLE
Add getAll method to UserDao

### PR DESCRIPTION
**Pull request by mkuznyetsov**
### New Behavior

Add getAll method to UserManager/UserDao for retrieving users with pagination
### Tests written?

Yes

Please review [Che's Contributing Guide](https://github.com/eclipse/che/CONTRIBUTING.MD) for best practices.

### CODE REVIEW

File: wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/UserManager.java [hunk 1]
```diff
@@ -208,6 +210,25 @@ public User getByEmail(String email) throws NotFoundException, ServerException {
     }
 
     /**
+     * Finds user by given {@code email}.
```
Comment (sleshchenko) [inline wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/UserManager.java]: Check doc

**Comment by skabashnyuk** [comment]
ok
@evoevodin @akorneta ?

File: wsmaster/che-core-api-user/src/test/java/org/eclipse/che/api/user/server/UserManagerTest.java [hunk 2]
```diff
@@ -203,6 +205,28 @@ public void shouldGetUserByEmail() throws Exception {
         assertEquals(manager.getByEmail(user.getEmail()), user);
     }
 
+    @Test
+    public void shouldGetAllUsers() throws Exception {
```
Comment (sleshchenko) [inline wsmaster/che-core-api-user/src/test/java/org/eclipse/che/api/user/server/UserManagerTest.java]: Should we test that paging works correctly?

File: wsmaster/che-core-api-user/src/test/java/org/eclipse/che/api/user/server/spi/tck/UserDaoTest.java [hunk 3]
```diff
@@ -183,6 +186,41 @@ public void shouldThrowNpeWhenGettingUserByNullAlias() throws Exception {
         userDao.getByAlias(null);
     }
 
+    @Test
+    public void getAllShouldReturnAllUsersWithinSingleResponse() throws Exception {
```
Comment (sleshchenko) [inline wsmaster/che-core-api-user/src/test/java/org/eclipse/che/api/user/server/spi/tck/UserDaoTest.java:190]: Should we test that paging works correctly?

File: wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/jpa/JpaUserDao.java [hunk 4]
```diff
@@ -160,6 +164,32 @@ public UserImpl getByEmail(String email) throws NotFoundException, ServerExcepti
...
+            return new Page<>(list, skipCount, maxItems, getTotalCount());
+        } catch (RuntimeException x) {
+            throw new ServerException(x.getLocalizedMessage(), x);
+        }
+    }
+
+    @Transactional
+    protected int getTotalCount() throws ServerException {
+        try {
+            return ((Number) managerProvider.get().createNamedQuery("User.getTotalCount").getSingleResult()).intValue();
```
Comment (voievodin) [inline wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/jpa/JpaUserDao.java]: use Long.class as second argument in `.createNamedQuery` method, and please remove the cast

File: wsmaster/che-core-api-user/src/test/java/org/eclipse/che/api/user/server/spi/tck/UserDaoTest.java (hunk 3 above)
Comment (mkuznyetsov) [inline wsmaster/che-core-api-user/src/test/java/org/eclipse/che/api/user/server/spi/tck/UserDaoTest.java:190]: ok, added test

File: wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/jpa/JpaUserDao.java (hunk 4 above)
Comment (mkuznyetsov) [inline wsmaster/che-core-api-user/src/main/java/org/eclipse/che/api/user/server/jpa/JpaUserDao.java]: k

**Comment by sleshchenko** [comment]
LGTM

**Comment by skabashnyuk** [comment]
ок

**Comment by akorneta** [comment]
ok

**Comment by voievodin** [comment]
ok

**Pull request merged**